typing/typecore.ml

```diff
@@ -733,24 +733,25 @@ and build_as_type_aux env p =
         else p.pat_mode
       in
       let ty =
-        newty (Tvariant{row_fields=[l, Rpresent ty]; row_more=newvar();
+        newty (Tvariant{row_fields=[l, Rpresent ty];
+                        row_more=newvar Type_layout.value;
                         row_bound=(); row_name=None;
                         row_fixed=None; row_closed=false})
       in
       ty, mode
   | Tpat_record (lpl,_) ->
       let lbl = snd3 (List.hd lpl) in
       if lbl.lbl_private = Private then p.pat_type, p.pat_mode else
-      let ty = newvar () in
-      let ppl = List.map (fun (_, l, p) -> l.lbl_pos, p) lpl in
+      let ty = newvar Type_layout.any in
```

Comment:
Right: the call to `unify_pat` unifies this with `ty_res`, the type of the record.  I've left a comment.  If in the future we wanted to allow records types to have layout any and check on construction that they were sane records, this would need to change, but I don't think that's the case?